server/engine/rooms.js

```diff
@@ -7,39 +9,52 @@ function hash(x) {
   return h;
 }
 
+const nameLen = 4;
+
 class Rooms {
-  constructor(len = 4) {
-    this.len = len;
+  constructor() {
     this.count = Math.floor(Math.random() * 9001) | 0;
     this.rooms = {};
   }
 
   nextName() {
     let h = hash(this.count);
     const name = [];
-    for (let i = 0; i < this.len; i += 1) {
+    for (let i = 0; i < nameLen; i += 1) {
       name.push(String.fromCharCode("A".charCodeAt(0) + (h % 26)));
-      h /= 26;
+      h = (h / 26) | 0;
     }
     this.count += 1;
     return name.join("");
   }
 
-  createRoom(username) {
+  createRoom(userId) {
+    const user = users.getUser(userId);
+    if (!user) return null;
     const name = this.nextName();
-    const room = { name, creator: username, users: [username] };
+    const room = { name, users: [userId] };
```

Comment:
Any motivation for removing 'creator' from this room? Maybe 'admin' would be a better name, but I think we want a user who can kick other people from the room, do special things on a per game basis, etc.

Reply:
resolved in discussion about 0 index = leader